packages/sandcastle/src/Gallery/GalleryItemStore.ts

```diff
@@ -59,96 +49,110 @@ export function useGalleryItemStore() {
   // Searches
   const [searchTerm, setSearchTerm] = useState<string | null>(null);
   const [searchFilter, setSearchFilter] = useState<GalleryFilter>(null);
-  const [searchResults, setSearchResults] = useState<
-    PagefindSearchFragment[] | null
-  >(null);
+  const [searchResults, setSearchResults] = useState<VectorSearchResult[] | null>(null);
   const [isSearchPending, startSearch] = useTransition();
+  
+  // Perform vector search when search term changes (with debounce)
   useEffect(() => {
-    const pagefind = getPagefind();
```

Comment:
I don't think we want to completely remove pagefind so we still support offline or isolated environments